### PR TITLE
SDIT-2320 Cpnsider end dated bookings that are active as not really active

### PR DESCRIPTION
**Pull request by andymarke**
This might be after a data fix

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/contactperson/ContactPersonService.kt [hunk 1]
```diff
@@ -175,18 +175,17 @@ class ContactPersonService(
   fun findPersonIdsByFilter(
     pageRequest: Pageable,
     personFilter: PersonFilter,
-  ): Page<PersonIdResponse> =
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/contactperson/ContactPersonService.kt:178]: hmm - ktlint formating

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt [hunk 2]
```diff
@@ -25,66 +25,59 @@ class PrisonerService(
   private val mergeTransactionRepository: MergeTransactionRepository,
   private val offenderBookingRepository: OffenderBookingRepository,
 ) {
-  fun findAllActivePrisoners(pageRequest: Pageable): Page<PrisonerIds> =
-    bookingRepository.findAll(ActiveBookingsSpecification(), pageRequest)
-      .map { PrisonerIds(bookingId = it.bookingId, offenderNo = it.offender.nomsId) }
+  fun findAllActivePrisoners(pageRequest: Pageable): Page<PrisonerIds> = bookingRepository.findAll(ActiveBookingsSpecification(), pageRequest)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt:28]: more formatting

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/specification/ActiveBookingsSpecification.kt [hunk 3]
```diff
@@ -5,17 +5,28 @@ import jakarta.persistence.criteria.CriteriaQuery
...
     root: Root<OffenderBooking>,
-    query: CriteriaQuery<*>,
+    @Nullable query: CriteriaQuery<*>?,
     criteriaBuilder: CriteriaBuilder,
   ): Predicate? {
     val predicates = mutableListOf<Predicate>()
 
     predicates.add(criteriaBuilder.equal(root.get<String>(OffenderBooking::active.name), true))
+    // ignore prisoners that are out but with a booking end date - since the data must be in a bad state
+    // due to NOMIS data fix
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/specification/ActiveBookingsSpecification.kt:21]: the actual change

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/specification/ActiveBookingsSpecification.kt [hunk 4]
```diff
@@ -5,17 +5,28 @@ import jakarta.persistence.criteria.CriteriaQuery
...
 import jakarta.persistence.criteria.Root
 import org.springframework.data.jpa.domain.Specification
+import org.springframework.lang.Nullable
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderBooking
 
 class ActiveBookingsSpecification : Specification<OffenderBooking> {
   override fun toPredicate(
     root: Root<OffenderBooking>,
-    query: CriteriaQuery<*>,
+    @Nullable query: CriteriaQuery<*>?,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/specification/ActiveBookingsSpecification.kt:14]: fix kotlin compiler warning

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt (hunk 2 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt:28]: Actually - everything in this file is ktling changes - so not sure what has happened here - I did download a new version of the plugin

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**